bitcoin_executor/sources/interpreter.move

```diff
@@ -53,34 +87,127 @@ public fun isSuccess(ip: &Interpreter): bool {
     cast_to_bool(&top)
 }
 
-
 fun cast_to_bool(v: &vector<u8>): bool {
     let mut i = 0;
     while (i < v.length()) {
         if (v[i] != 0) {
-        // Can be negative zero
-             if (i == v.length()-1 && v[i] == 0x80)
-                return false;
+            // Can be negative zero
+            if (i == v.length()-1 && v[i] == 0x80) return false;
             return true;
         };
         i = i + 1;
     };
     false
 }
 
+fun op_push_empty_vector(ip: &mut Interpreter) {
+    ip.stack.push(vector[]);
+}
+
+fun op_push_n_bytes(ip: &mut Interpreter, num_bytes_to_push: u8) {
+    let data_to_push = ip.reader.read(num_bytes_to_push as u64);
+    ip.stack.push(data_to_push);
+}
+
+fun op_push_small_int(ip: &mut Interpreter, opcode: u8) {
+    // OP_1 (0x51) corresponds to 1  (0x51 - 0x50 = 0x01)
+    // OP_16 (0x60) corresponds to 16 (0x60 - 0x50 = 0x10)
+    let numeric_value: u8 = opcode - 0x50;
+    ip.stack.push(vector[numeric_value]);
+}
 
 // OP_DUP eval
 fun op_dup(ip: &mut Interpreter) {
     let value = ip.stack.top();
     ip.stack.push(value)
 }
 
+#[test]
+fun test_op_0() {
+    let stack = stack::create();
+    let mut ip = new(stack);
+    ip.op_push_empty_vector();
+
+    assert!(ip.stack.size() == 1);
+    let top_val = ip.stack.top();
+    assert!(top_val.length() == 0);
+    assert!(!ip.isSuccess());
```

Comment:
We don't need to test the `isSuccess` in this case. We only care about the state of the stack.

Reply:
its a extra check to see the value there is 0